### PR TITLE
Add annonce

### PR DESCRIPTION
**Pull request by Sofiaboud**
Routes backend for add annonce

### CODE REVIEW

**Review by ThomasSupervil — DISMISSED** [review]
seems good

**Comment by LoicBrassart** [comment]
SM validation before you call for me plz!

**Review by ThomasSupervil — DISMISSED** [review]
good now

File: package.json [hunk 1]
```diff
@@ -2,14 +2,15 @@
...
   "version": "1.0.0",
   "description": "",
-  "main": "index.js",
+  "main": "app.js",
   "keywords": [],
   "author": "",
   "license": "ISC",
   "scripts": {
-    "start": "node index.js",
+    "start": "nodemon app.js",
```
Comment (LoicBrassart) [inline package.json]: Nope (won't work once deployed)

File: package.json [hunk 2]
```diff
@@ -28,7 +29,9 @@
     "cors": "^2.8.5",
     "dotenv": "^9.0.2",
     "express": "^4.17.1",
-    "mysql2": "^2.2.5"
+    "joi": "^17.4.0",
+    "mysql2": "^2.2.5",
+    "nodemon": "^2.0.7"
```
Comment (LoicBrassart) [inline package.json]: Nodemon should be a **dev**Dependancies

File: models/annonce.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    animals: Joi.boolean().default(false),
+    title: Joi.string().max(45).required(),
+    rooms: Joi.number().integer(),
+    energyClass: Joi.string().max(1).required(),
+    description: Joi.string().max(1000).required(),
+    property: Joi.string().max(45).required(),
+    idUser: Joi.number().integer(),
+    ges: Joi.string().max(1),
+    surface: Joi.number(),
+  }).validate(body, { abortEarly: false }).error;
```
Comment (LoicBrassart) [inline models/annonce.js]: If you wanna use `Joi`, I want your backend validation criteria to match your db requirements exactly (types of fields, data length, etc) on the whole project, and every field transmitted by the frontend bia any mean will have to go through a strict Joi validation process.
I'll take the merging of this PR into `dev` as a collective (both groups!) agreement to abide by this new standard

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Some Evils and a decision to make

**Review by ThomasSupervil — DISMISSED** [review]
✌️

**Comment by LoicBrassart** [comment]
Solve conflicts before you ask for my review please

**Review by ThomasSupervil — DISMISSED** [review]
resolve conflict

**Review by ThomasSupervil — DISMISSED** [review]
good

File: models/annonce.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
+const { db } = require('../conf');
+
+const getAll = () => {
+  const sql = 'SELECT * FROM accomodations';
```
Comment (LoicBrassart) [inline models/annonce.js]: No "SELECT *"

File: models/annonce.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  return db.query(sql);
+};
+
+const getOne = (id) => {
+  const sqlValues = [id];
+  const sql = 'SELECT * FROM accomodations WHERE id = ?';
```
Comment (LoicBrassart) [inline models/annonce.js]: No "SELECT *"

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Don't use "SELECT *"

**Review by ThomasSupervil — APPROVED** [review]
No "SELECT *" at the horizon

**Review by LoicBrassart — APPROVED** [review]
SnE

**Pull request merged**